app/src/main/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt

```diff
@@ -31,17 +31,37 @@ class WelcomeFragment : Fragment() {
     private lateinit var rootView: View
     private val geoLocationViewModel by viewModel<GeoLocationViewModel>()
     val preference = Preference()
+    private lateinit var editor: SharedPreferences.Editor
+
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         rootView = inflater.inflate(R.layout.fragment_welcome, container, false)
         val thisActivity = activity
         if (thisActivity is AppCompatActivity)
             thisActivity.supportActionBar?.hide()
+
         rootView.pickCityButton.setOnClickListener {
             Navigation.findNavController(rootView)
                 .navigate(WelcomeFragmentDirections.actionWelcomeToSearch(WELCOME_FRAGMENT))
         }
 
+
+        when (preference.getBoolean("dialog", false)) {
+            false -> AlertDialog.Builder(thisActivity!!)
+                .setTitle("Message")
+                .setMessage(R.string.dialog_message).setPositiveButton("Okay") { dialog, it ->
+                    preference.putBoolean("dialog",false)
```

Comment:
I don't get it. You show the dialog if the preference is false, and then set the preference to false when the user clicks OK?